### ISSUE TITLE
GetUserInfo 401 error

### ISSUE THREAD

**Issue by mobinseven** [enkodellc/blazorboilerplate#102]
Hi. 
**Summary:**
A 401 error is returned by `GetUserInfo` when user in not logged in. Causes a dummy browser login request (in some cases).
**Steps to reproduce:**
Browse the app without logging in.
**Resolution:**
The `GetUserInfo` in `IdentityAuthenticationStateProvider` class is requesting to get the user info even though it knows user is not logged in. So, I propose this structure for  `GetUserInfo`:
```
        public async Task<UserInfoDto> GetUserInfo()
        {            
            if (_userInfoCache != null && _userInfoCache.IsAuthenticated)
            {
                _userInfoCache = await _authorizeApi.GetUserInfo();
                return _userInfoCache;
            }

            //If the user is not authenticated then an empt UserInfoDto is returned
            //_userInfoCache = await _authorizeApi.GetUserInfo();
            return new UserInfoDto();
        }
```

**Comment by enkodellc** [comment]
@mobinseven Thanks I agree there might be a better solution. But implementing your suggestion causes a side effect. The login doesn't work after implementation. 

I think the real is removing the [Authorize] from the server API call _api/Account/UserInfo_  and returning an Anonymous UserProfile.  I think it was accidentally added during the IS4 refactor. I will push that change.

**Comment by mobinseven** [comment]
Yes. I should've tested that. Maybe we can fill `_userInfoCache` somewhere else (ugly solution?):
`IdentityAuthenticationStateProvider`:
```
        public async Task<ApiResponseDto> Login(LoginDto loginParameters)
        {
            ApiResponseDto apiResponse = await _authorizeApi.Login(loginParameters);
            if (apiResponse.StatusCode == 200)
            {
                _userInfoCache = await _authorizeApi.GetUserInfo();
            }
            NotifyAuthenticationStateChanged(GetAuthenticationStateAsync());
            return apiResponse;
        }
```

**Comment by mobinseven** [comment]
The above code has a problem too: when browser page is refreshed, _userInfoCache will be cleared; but user is still logged in. I did a bunch of work and solved this issue:

1. `GetUser` reworked: Added `GetUser` to `IAuthorizeApi`. Removed its `[Authorize]` attribute in server. Made it the main reference of authentication state.
2. Did a coup d'état against `_userInfoCache`. removed it from the cycle.
3. Replaced `GetAuthenticationStateAsync` calls with `GetUserInfo` in many cases.

And so far found no problem in that structure.
